lib/griddler/mailgun/adapter.rb

```diff
@@ -80,12 +82,26 @@ def param_or_header(key)
       end
 
       def attachment_files
-        attachment_count = params['attachment-count'].to_i
-
-        attachment_count.times.map do |index|
-          params.delete("attachment-#{index+1}")
+        attachments = JSON.parse(params["attachments"]) || Array.new
+        attachments.map do |attachment|
+          ActionDispatch::Http::UploadedFile.new(
+            {
+              filename: attachment["name"],
+              type: attachment["content-type"],
+              tempfile: create_tempfile(attachment)
+            }
+          )
         end
       end
+
+      def create_tempfile(attachment)
+        tempfile = Tempfile.new(attachment["name"])
+        tempfile.binmode
+        tempfile << open(attachment["url"], 
+          :http_basic_authentication => ["api", Rails.configuration.apikey]).read
```

Comment:
Line is too long. [81/80]<br>Align the parameters of a method call if they span more than one line.<br>Use the new Ruby 1.9 hash syntax.